### PR TITLE
#111 fix etl tests

### LINKED ISSUE #111: Fix the pytests for pipeline
**Description**
Amend all the test files in pipeline so that all of the tests will pass

**Outcome**
All the tests will pass

**User story**
As a developer, I want to amend all the test files in the pipeline, so that all tests will pass successfully, ensuring the integrity and reliability of the codebase.

### PR DESCRIPTION
**Pull request by keoghrmj**
All tests are now fixed and they all now pass

Closes #111

### CODE REVIEW

File: pipeline/test_load.py [hunk 1]
```diff
@@ -127,29 +137,25 @@ def test_keyword_already_exists_no_insert(mock_setup):
...
-    assert mock_curs.execute.call_count == 3
-    assert mock_conn.commit.call_count == 3
+    insert_keyword_recordings(mock_conn, mock_curs, mock_df_2)
 
     mock_curs.execute.assert_any_call(
         """INSERT INTO keyword_recordings
-                       (keywords_id, total_mentions, avg_sentiment, hour_of_day, recorded_at)
-                       VALUES (%s, %s, %s, %s, %s)""",
-        (3, 18, 0.80, '10:00', mock_recorded_at)
+                       (keywords_id, total_mentions, avg_sentiment, date_and_hour)
```
Comment (Kurt812) [inline pipeline/test_load.py:152]: Nice update

**Review by Kurt812 — COMMENTED** [review]
Nice changes, cleaning up the code

File: rds_to_s3/test_etl.py [hunk 2]
```diff
@@ -153,14 +154,13 @@ def test_unsuccessful_s3_connection_missing_env(mock_client, caplog):
 @patch('boto3.client')
 def test_s3_connection_raises_client_error(mock_client, caplog):
     """Test that s3 connection function can and will raise the NoCredentialsError."""
-    mock_client.side_effect = ClientError( error_response={'Error': {'Code': 'AuthFailure', 'Message': 'Authentication failure'}},
-        operation_name='connect')
+    mock_client.side_effect = ClientError(error_response={'Error': {'Code': 'AuthFailure', 'Message': 'Authentication failure'}},
+                                          operation_name='connect')
```
Comment (SurinaCS) [inline rds_to_s3/test_etl.py:158]: great stuff

File: rds_to_s3/test_etl.py [hunk 3]
```diff
@@ -372,7 +373,8 @@ def test_upload_s3_exception(mock_s3_conn, caplog):
     mock_s3.upload_file.assert_called_once_with(
         f'/tmp/{file_name}', 'test_bucket', 'object_name')
 
-    assert f'Unexpected error while uploading {file_name} to S3: ' in caplog.text
+    assert f'Unexpected error while uploading {
+        file_name} to S3: ' in caplog.text
```
Comment (SurinaCS) [inline rds_to_s3/test_etl.py:377]: nice

**Review by SurinaCS — APPROVED** [review]
well done Keogh

File: pipeline/api.py [hunk 4]
```diff
@@ -21,5 +21,5 @@ def add_topic()-> None:
 
 
 if __name__ == "__main__":
-    # app.run(host="0.0.0.0", port=5000, debug=True)
-    app.run(debug=True)
+    app.run(host="0.0.0.0", port=5000, debug=True)
+    # app.run(debug=True)
```
Comment (Kurt812) [inline pipeline/api.py:25]: could we remove this comment?

**Pull request merged**